x-pack/plugins/maps/public/routing/routes/maps_app/load_map_and_render.tsx

```diff
@@ -5,15 +5,31 @@
  */
 
 import React from 'react';
-import { MapsAppView } from '.';
-import { getMapsSavedObjectLoader } from '../../bootstrap/services/gis_map_saved_object_loader';
-import { getCoreChrome, getToasts } from '../../../kibana_services';
 import { i18n } from '@kbn/i18n';
 import { Redirect } from 'react-router-dom';
+import { AppMountParameters } from 'kibana/public';
+import { EmbeddableStateTransfer } from 'src/plugins/embeddable/public';
+import { getCoreChrome, getToasts } from '../../../kibana_services';
+import { getMapsSavedObjectLoader } from '../../bootstrap/services/gis_map_saved_object_loader';
+import { MapsAppView } from '.';
+import { ISavedGisMap } from '../../bootstrap/services/saved_gis_map';
+
+interface Props {
+  savedMapId?: string;
+  onAppLeave: AppMountParameters['onAppLeave'];
+  stateTransfer: EmbeddableStateTransfer;
+  originatingApp?: string;
+}
+
+interface State {
+  savedMap?: ISavedGisMap;
+  failedToLoad: boolean;
+}
 
-export const LoadMapAndRender = class extends React.Component {
+export const LoadMapAndRender = class extends React.Component<Props, State> {
+  _isMounted!: boolean;
```

Comment:
Instead of making optional, just init to false

Reply:
Fixed this.